pkg/webhooks/operator_webhooks.go

```diff
@@ -147,6 +173,65 @@ func (webhookConfig *IntegreatlyWebhookConfig) Reconcile(ctx context.Context, cl
 	return nil
 }
 
+// ReconcileService creates or updates the service that points to the Pod
+func (webhookConfig *IntegreatlyWebhookConfig) ReconcileService(ctx context.Context, client k8sclient.Client) error {
+	service := &corev1.Service{
+		ObjectMeta: v1.ObjectMeta{
+			Name:      operatorPodServiceName,
+			Namespace: "redhat-rhmi-operator",
+		},
+	}
+
+	_, err := controllerutil.CreateOrUpdate(ctx, client, service, func() error {
```

Comment:
Won't this fail when the service already exists because we're not setting the clusterIP? Or is the old value of the clusterIP implicitely copied here? What about always setting it to `None`?

Reply:
It probably won't fail createOrUpdate applies changes to the object it first `Get`s.
So the clusterIP field in the `Update` call will be set to previous value.

But I raised a related concern in [a comment in JIRA](https://issues.redhat.com/browse/INTLY-7829?focusedCommentId=14108855&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-14108855), so we might need to set it to `None` for that purpose.

Reply:
No, it will keep whatever value it had before, or leave it empty it it's being created. The problem of setting it to `None` is that if the existing service has a different value it will fail.
This way the upgrade from 2.2.0 won't fail as the existing service has an automatically assigned clusterIP

Reply:
We can do Get before the CreateOrUpdate, and Delete Service if clusterIP is not empty.
This would get all clusters into a consistent state.
@sergioifg94 @pb82  Wdyt?

Reply:
That sounds good to me. Deleting it would give us more control over the clusterIP